graylog2-web-interface/src/stores/inputs/InputStatesStore.js

```diff
@@ -28,13 +28,109 @@ const InputStatesStore = Reflux.createStore({
             }
             result[input.id][node] = input;
           });
-        })
+        });
         this.inputStates = result;
         this.trigger({inputStates: this.inputStates});
 
         return result;
       });
-  }
+  },
+
+  start(input) {
+    let url;
+
+    if (input.global) {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.ClusterInputStatesController.start(input.id).url);
+    } else {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.InputStatesController.start(input.id).url);
+    }
+
+    const handleSuccess = () => UserNotification.success(`Input '${input.title}' started successfully`);
+    const handleError = (message) => UserNotification.error(message, `Input '${input.title}' could not be started`);
+
+    return fetch('PUT', url)
+      .then(
+        () => {
+          this.list().then(inputStatesList => this._checkStartedState(input, inputStatesList, handleSuccess, handleError));
+        },
+        error => {
+          handleError(`Error starting input '${input.title}': ${error}`);
+        });
+  },
+
+  stop(input) {
+    let url;
+
+    if (input.global) {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.ClusterInputStatesController.stop(input.id).url);
+    } else {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.InputStatesController.stop(input.id).url);
+    }
+
+    const handleSuccess = () => UserNotification.success(`Input '${input.title}' stopped successfully`);
+    const handleError = (message) => UserNotification.error(message, `Input '${input.title}' could not be stopped`);
+
+    return fetch('DELETE', url)
+      .then(
+        () => {
+          this.list().then(inputStatesList => this._checkStoppedState(input, inputStatesList, handleSuccess, handleError));
```

Comment:
I am still not happy about this. My comment was targeted towards the results of the individual start/stop calls. As input launching/stopping is done asynchronously, we cannot check for the success of the whole operation, but we can (and should) check if _triggering_ them is successful. 

The way it is done now implies that the user is notified only when input startup/termination succeeds/fails during the short time span of sending the request which triggers those and finishing the `list` operation. This might be too short for some inputs. So my suggestion is:

1) Just check for the individual success/failure of the operations triggering async start/stop and indicate the result (all succeeded, mixed, all failed) to the user.
2) Signal status of input states separately using badges in input list (what we do already)